### ISSUE TITLE
Barrier node on junction leading to incorrect blocking behaviour

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper#388]
In rare cases barrier nodes can be passed: https://graphhopper.com/maps/?point=51.240124%2C7.149857&point=51.240154%2C7.150313

See:  http://www.openstreetmap.org/node/1217545630

@NopMap shouldn't a barrier on a node block access from and to every edge? I mean in this example it is obviously wrong but the tagging would block everything and is also incorrect, right?

**Comment by jansoe** [comment]
> In rare cases barrier nodes can be passed
to be more specific, barrier nodes always can be passed if there are more than one way connected to them

**Comment by NopMap** [comment]
Correct. This is invalid tagging. See the OSM wiki http://wiki.openstreetmap.org/wiki/Barriers

> Tagging: place a node at the location of the barrier on an existing way (either a highway or a linear
> barrier). Do not place a node-barrier at the junction of two highways, else it will restrict the traffic on
> both highways. 

The gate should only be put onto the way it is actually blocking, close to the end node.

**Comment by karussell** [comment]
@jansoe was this fixed or do we assume invalid tagging and close?

**Comment by jansoe** [comment]
not yet fixed (only in branch trafficlights2).
It's invalid tagging, but I would vote to employ it nevertheless for 2 reasons
(1) First the wrong tagging becomes more obvious
(2) Currently, blocked ways might become passable in GraphHopper, so people might become stuck. With the implementation of the wrong tagging, no one gets stuck (only people might have to take a detour)
